src/store/fs.rs

```diff
@@ -0,0 +1,641 @@
+//! Provides a JSON file-based implementation for the caching system.
+
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use super::errors::StoreError;
+use crate::model::{PacBuild, Repository};
+use crate::store::base::{Base, StoreResult};
+use crate::store::filters::{InstallState, Kind};
+
+#[cfg(not(test))]
+const FSS_PATH: &str = "/etc/pacstall/fss.json";
+#[cfg(test)]
+const FSS_PATH: &str = "./fss.json";
+
+/// `FileSystem` implementation for the caching system
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct FileSystemStore {
+    repositories: Vec<Repository>,
+    packages: HashMap<String, Vec<PacBuild>>,
+
+    #[serde(skip_serializing)]
+    allow_data_save: bool,
+}
+
+impl FileSystemStore {
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new() -> Box<dyn Base> {
+        Box::new(FileSystemStore {
+            repositories: Vec::new(),
+            packages: HashMap::new(),
+            allow_data_save: true,
+        })
+    }
+
+    /// # Private
+    fn get_packages_by_repository(
+        &mut self,
+        repository_url: &str,
+    ) -> Result<&mut Vec<PacBuild>, StoreError> {
```

Comment:
```suggestion
    ) -> StoreResult<&mut Vec<PacBuild>> {
```

Use the type alias wherever appropiate.